trajopt_ifopt/include/trajopt_ifopt/constraints/cartesian_position_constraint.h

```diff
@@ -0,0 +1,156 @@
+    : manip_(std::move(manip))
+    , adjacency_map_(std::move(adjacency_map))
+    , world_to_base_(world_to_base)
+    , link_(std::move(link))
+    , tcp_(tcp)
+  {
+    kin_link_ = adjacency_map_->getLinkMapping(link_);
+    if (kin_link_ == nullptr)
+    {
+      CONSOLE_BRIDGE_logError("Link name '%s' provided does not exist.", link_.c_str());
+      assert(false);
+    }
+  }
+
+  tesseract_kinematics::ForwardKinematics::ConstPtr manip_;
+  tesseract_environment::AdjacencyMap::ConstPtr adjacency_map_;
+  Eigen::Isometry3d world_to_base_;
+  std::string link_;
+  tesseract_environment::AdjacencyMapPair::ConstPtr kin_link_;
+  Eigen::Isometry3d tcp_;
+};
+
+class CartPosConstraint : public ifopt::ConstraintSet
+{
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  using Ptr = std::shared_ptr<CartPosConstraint>;
+  using ConstPtr = std::shared_ptr<const CartPosConstraint>;
+
+  CartPosConstraint(const Eigen::Isometry3d& target_pose,
+                    CartPosKinematicInfo::ConstPtr kinematic_info,
+                    JointPosition::Ptr position_var,
+                    const std::string& name = "CartPos");
+
+  /**
+   * @brief CalcValues Calculates the values associated with the constraint
+   * @param joint_vals Input joint values for which FK is solved
+   * @return Error of FK solution from target, size 6. The first 3 terms are associated with position and the last 3 are
+   * associated with orientation.
+   */
+  Eigen::VectorXd CalcValues(const Eigen::Ref<Eigen::VectorXd>& joint_vals) const;
+  /**
+   * @brief Returns the values associated with the constraint. In this case that is the concatenated joint values
+   * associated with each of the joint positions should be n_dof_ * n_vars_ long
+   * @return
+   */
+  Eigen::VectorXd GetValues() const override;
+
+  /**
+   * @brief  Returns the "bounds" of this constraint. How these are enforced is up to the solver
+   * @return Returns the "bounds" of this constraint
+   */
+  std::vector<ifopt::Bounds> GetBounds() const override;
+
+  void SetBounds(const std::vector<ifopt::Bounds>& bounds);
+
+  /**
+   * @brief Fills the jacobian block associated with the constraint
+   * @param jac_block Block of the overall jacobian associated with these constraints
+   */
+  void CalcJacobianBlock(const Eigen::Ref<Eigen::VectorXd>& joint_vals, Jacobian& jac_block) const;
+  /**
+   * @brief Fills the jacobian block associated with the given var_set.
+   * @param var_set Name of the var_set to which the jac_block is associated
+   * @param jac_block Block of the overal jacobian associated with these constraints and the var_set variable
+   */
+  void FillJacobianBlock(std::string var_set, Jacobian& jac_block) const override;
+
+  void SetTargetPose(const Eigen::Isometry3d& target_pose);
+
+  const CartPosKinematicInfo::ConstPtr& getKinematicInfo() { return kinematic_info_; }
+
+  Eigen::Isometry3d GetTargetPose() const;
+
+  Eigen::Isometry3d GetCurrentPose() const;
```

Comment:
Documentation Missing for methods above.

Reply:
Done